obj.go

```diff
@@ -87,6 +87,24 @@ func (O *Object) SetAttribute(name string, data *Data) error {
 	return nil
 }
 
+// InitAttributes prepare all atributes for use the object as IN parameter
+func (O *Object) InitAttributes() error {
+	var data Data
+	for _, attr := range O.Attributes {
+		data.reset()
+		data.NativeTypeNum = attr.NativeTypeNum
+		data.ObjectType = attr.ObjectType
+		if attr.NativeTypeNum == C.DPI_NATIVE_TYPE_BYTES && attr.OracleTypeNum == C.DPI_ORACLE_TYPE_NUMBER {
```

Comment:
Ok, I'm still confused.

The comment says you want to bind the attributes as IN parameter - so you want to send the Object's contained data to the DB.
That's what the use of dpiObject_setAttributeValue supports, too.

But if that's the case, dpiData_setBytes was the good choice, sorry.
Also the use of the zero array...

So please revert the change, and put dpiData_setBytes back.

And either document that it resets the attributes or rename the function to "Reset", instead of InitAttributes.
Or ResetAttributes.